File: src/webapp/client/src/ts/types/risks.ts
export interface RawRisk {
    Id                    : number
    Name                  : string
    HumanId               : string
    EngagementId          : number
    Severity              : number
    SeverityJustification : string
    Description           : string
}

export function createEmptyRisk() : RawRisk {
    return {
        Id: -1,
        Name: '',
        HumanId: '',
        EngagementId: -1,
        Severity: 0,
        SeverityJustification: '',
        Description: '',
    }
}

File: src/shared/golang/backend/risks/risks.go
package risks

import (
	"github.com/jmoiron/sqlx"
)

type Risk struct {
	Id                    int64  `db:"id"`
	EngagementId          int64  `db:"engagement_id"`
	Name                  string `db:"name"`
	HumanId               string `db:"human_id"`
	Severity              int16  `db:"severity"`
	SeverityJustification string `db:"severity_justification"`
	Description           string `db:"description"`
}

type RiskManager struct {
	db *sqlx.DB
}

func CreateRiskManager(db *sqlx.DB) *RiskManager {
	return &RiskManager{
		db: db,
	}
}
